src/common/core-functions.k

```diff
@@ -109,14 +139,35 @@ syntax KItem ::=  mapDiff(
                 K   //setWrap(Set)
               )
               [strict]
-
-rule mapDiff( mapWrap( (Key:K |-> _ => .Map) _), setWrap( (SetItem(Key) => .Set) _) )
-
-rule mapDiff( mapWrap( MyMap:Map ), setWrap( (SetItem(Key:K) => .Set) _) )
-when
-    notBool Key in keys(MyMap)
-
-rule mapDiff(mapWrap( MyMap:Map ), setWrap(.Set)) => mapWrap(MyMap)
+//todo
+//rule mapDiff( mapWrap( (Key:K |-> _ => .Map) _:Map), setWrap( (SetItem(Key) => .Set) _:Set) )
+//rule mapDiff( mapWrap( MyMap:Map ), setWrap( (SetItem(Key:K) => .Set) _:Set) )
+//when    notBool Key in keys(MyMap)
+//rule mapDiff(mapWrap( MyMap:Map ), setWrap(.Set)) => mapWrap(MyMap)
+
+//@ Substracts from the given map the value given (match is on the value)
+syntax KItem ::= removeMapItemsByValue (Map,K)             [function]
```

Comment:
At first I thought I could, but when I changed it to Map, I got warnings

[Warning] Compiler: unable to transform the KApp:
'removeMapItemsByValue(M:Map,, V:K,, org.kframework.kil.MapBuiltin@64bb137)
because of multiple productions associated:
["removeMapItemsByValue" "(" Map "," K ")" , "removeMapItemsByValue" "(" K ","
K "," Map "," K "," Map ")" , "removeMapItemsByValue" "(" Map "," K "," Map ")"
]